base/MAPL_ObsUtil.F90

```diff
@@ -326,45 +326,140 @@ subroutine read_M_files_4_swath ( filenames, Xdim, Ydim, &
+    if ( present(Tfilter) .AND. Tfilter ) then
+       if ( .not. (present(time) .AND. present(lon) .AND. present(lat)) ) then
+          _FAIL('when Tfilter present, time/lon/lat must also present')
+       end if
+
+       !
+       ! -- determine jx
+       !
+       jx=0
+       do i = 1, M
+          filename = filenames(i)
+          nlon = nlons(i)
+          nlat = nlats(i)
           allocate (time_loc_R8(nlon, nlat))
           call get_var_from_name_w_group (var_name_time, time_loc_R8, filename, _RC)
-          time(1:nlon,jx+1:jx+nlat) = time_loc_R8(1:nlon,1:nlat)
+!!          write(6,*) 'af ith, filename', i, trim(filename)
+
+          do j=1, nlat
+             !
+             ! -- filter, e.g., eliminate -9999
+             !
+             if ( time_loc_R8(1, j) > 0.0 ) then
+                jx = jx + 1
+             end if
+          end do
           deallocate(time_loc_R8)
+       end do
+       Xdim=nlon
+       Ydim=jx
+       if (allocated (time)) then
+          deallocate(time)
+          allocate (time(Xdim, Ydim))
        end if
-
-       if (present(var_name_lon).AND.present(lon)) then
+       if (allocated (lon)) then
+          deallocate(lon)
+          allocate (lon(Xdim, Ydim))
+       end if
+       if (allocated (lat)) then
+          deallocate(lat)
+          allocate (lat(Xdim, Ydim))
+       end if
+       !
+       !!write(6,'(2x,a,10i10)') 'true  Xdim, Ydim:', Xdim, Ydim
+       !!write(6,'(2x,a,10i10)') 'false Xdim, Ydim:', nlon, j2
+       !
+
+
+       !
+       ! -- determine true time/lon/lat by filtering T < 0
+       !
+       jx=0
+       do i = 1, M
+          filename = filenames(i)
+          nlon = nlons(i)
+          nlat = nlats(i)
+          !!write(6,'(2x,a,10i6)')  'M, i, nlon, nlat:', M, i, nlon, nlat
+          !!write(6,'(2x,a)') 'time_loc_r8'
+          !
+          allocate (time_loc_R8(nlon, nlat))
+          call get_var_from_name_w_group (var_name_time, time_loc_R8, filename, _RC)
           allocate (lon_loc(nlon, nlat))
           call get_var_from_name_w_group (var_name_lon, lon_loc, filename, _RC)
-          lon(1:nlon,jx+1:jx+nlat) = lon_loc(1:nlon,1:nlat)
-          deallocate(lon_loc)
-       end if
-
-       if (present(var_name_lat).AND.present(lat)) then
           allocate (lat_loc(nlon, nlat))
           call get_var_from_name_w_group (var_name_lat, lat_loc, filename, _RC)
-          lat(1:nlon,jx+1:jx+nlat) = lat_loc(1:nlon,1:nlat)
+          !
+          do j=1, nlat
+             !
+             ! -- filter, e.g., eliminate -9999
+             !
+             if ( time_loc_R8(1, j) > 0.0 ) then
+                jx = jx + 1
+                time(1:nlon,jx) = time_loc_R8(1:nlon,j)
+                lon (1:nlon,jx) = lon_loc (1:nlon,j)
+                lat (1:nlon,jx) = lat_loc (1:nlon,j)
+             end if
+             !!write(6,'(5f20.2)') time_loc_R8(1,j)
+          end do
+
+          !!write(6,'(2x,a,10i10)') 'end of file id', i
+          !!write(6,*)
```

Comment:
If you want to keep these prints for debugging purposes, it is probably time to learn a bit more about pFlogger.   That would avoid using commented code.

Reply:
Thanks. It is deleted in another coming PR.